src/app/storage/duck/operations.ts

```diff
@@ -181,20 +182,29 @@ function fetchMigStorageRefs(client: IClusterClient, migMeta, migStorages): Arra
       CoreNamespacedResourceKind.Secret,
       secretRef.namespace
     );
-    refs.push(client.get(secretResource, secretRef.name));
+    const resp = client.get(secretResource, secretRef.name);
+    resp.then(r => {
```

Comment:
we are hoping to standardize across the app using async/await or yield/generators & trying to avoid using legacy promise code.

Reply:
I'm thinking we probably want to deprecate thunks and legacy promise code unless for some reason they're absolutely necessary.